### DISCUSSION TITLE
Undo redo in joints free version

### DISCUSSION THREAD

**Discussion by adarshjalaja** [clientIO/joint#2273] [Q&A]
### Introduction

is it possible to do undo and redo in joint JS free version 

can we capture delta between two graphs and keep saving. is that a good approach

or do we need to save whole graph snapshot at each action and restore based on undo and redo action

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Duplicate of https://github.com/clientIO/joint/discussions/1654
